### ISSUE TITLE
[Research] Add EEGNet-Lite for On-Device Inference (15 KB, 15ms, INT8 quantized)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#67]
## What
Train a compact EEGNet (~2600 parameters) and quantize to INT8 for on-device mobile inference. Eliminates ML backend dependency for basic emotion classification.

## Why
- ACM ISWC (2024, ETH Zurich): EEGNet runs on GAP9 RISC-V with **15.6 KB memory, 14.9 ms inference, 0.76 mJ per inference**
- Online SGD fine-tuning of last dense layer adds **+7.31% accuracy** — personalization in 20 microseconds
- NeuralDreamWorkshop already has local ONNX inference in the mobile app — this provides a much smaller, faster model
- Current `emotion_classifier_model.onnx` is 2.2 MB — EEGNet-Lite would be <20 KB

### Step 1: Train EEGNet for 4 channels
```python
# ml/models/eegnet_lite.py
class EEGNetLite(nn.Module):
    def __init__(self, n_channels=4, n_classes=6, fs=256):
        # Temporal convolution: (1, 64) kernel
        # Depthwise spatial: (4, 1) per channel
        # Separable conv: (1, 16)
        # Average pooling + softmax
        # Total: ~2600 parameters
```

### Step 2: Quantize and export
```python
# PyTorch dynamic quantization → INT8
# Export to ONNX, TFLite, CoreML
```

### Step 3: Add personalization layer
```python
# Online SGD on last dense layer with user's labeled data
# 20 µs per update — runs on every user correction
```

### Files to Modify
- `ml/models/eegnet_lite.py` (NEW) — compact EEGNet architecture
- `ml/training/train_eegnet_lite.py` (NEW) — training + quantization
- `client/public/models/eegnet_lite.onnx` — exported quantized model

### Expected Impact
- Model size: <20 KB (vs 2.2 MB current ONNX)
- Inference: <15 ms (enables true real-time on mobile)
- Accuracy within 2-3% of full model
- Online personalization via last-layer SGD: +5-7%

## Complexity
LOW-MEDIUM — EEGNet architecture is simple, quantization tools are mature

## References
- [On-device Learning of EEGNet for Wearable Motor Imagery BCI](https://dl.acm.org/doi/10.1145/3675095.3676607) — ACM ISWC, 2024 (ETH Zurich)
- [ScheduledKD-LDC Knowledge Distillation for BCI](https://arxiv.org/abs/2403.13844) — arXiv, 2024
